### PR TITLE
Change Rx extension names

### LINKED ISSUE #436: Kotlin naming conventions violation in RxFuel
# Refactoring

## Description
Add to com.github.kittinunf.fuel.rx.RxFuel methods toRxRespone(), toRxResponseString() etc. Deprecate old methods.

## Why this is necessary?
Current naming violate Kotlin style: [link](https://kotlinlang.org/docs/reference/coding-conventions.html#function-names)

### PR DESCRIPTION
**Pull request by SleeplessByte**
## Description

Fixes #436

## Type of change

Check all that apply

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Refactoring (a change which changes the current internal or external interface)
- [X] This change requires a documentation update

## How Has This Been Tested?

In case you did not include tests describe why you and how you have verified the
changes, with instructions so we can reproduce. If you have added comprehensive
tests for your changes, you may omit this section.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation, if necessary
- [X] My changes generate no new compiler warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Inspect the bytecode viewer, including reasoning why

### CODE REVIEW

**Comment by NikkyAI** [comment]
for something as _simple_ as renaming functions
i'd recommend to keep the old names and deprecating them: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-deprecated/index.html

this will ensure projects depending on it can migrate (idea will allow autoreplace and quickfixes)

**Comment by SleeplessByte** [comment]
Yep, will deprecate the,.

File: fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt [hunk 1]
```diff
@@ -11,33 +12,137 @@ import com.github.kittinunf.result.Result
...
+)
+fun Request.rx_responseString(charset: Charset = Charsets.UTF_8) = rxResponseString(charset)
+
+@Suppress("FunctionName")
+@Deprecated(
+    "Use Request.rxResponseObject",
+    replaceWith = ReplaceWith("rxResponseObject(deserializable)"),
+    level = DeprecationLevel.ERROR
+)
+fun <T: Any> Request.rx_responseObject(deserializable: Deserializable<T>) = rxResponseObject(deserializable)
```
Comment (kittinunf) [inline fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt]: **[ktlint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Missing spacing before ":"

File: fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt [hunk 2]
```diff
@@ -11,33 +12,137 @@ import com.github.kittinunf.result.Result
...
+)
+fun Request.rx_string(charset: Charset = Charsets.UTF_8) = rxString(charset)
+
+@Suppress("FunctionName")
+@Deprecated(
+    "Use Request.rxObject",
+    replaceWith = ReplaceWith("rxObject(deserializable)"),
+    level = DeprecationLevel.ERROR
+)
+fun <T: Any> Request.rx_object(deserializable: Deserializable<T>) = rxObject(deserializable)
```
Comment (kittinunf) [inline fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt]: **[ktlint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Missing spacing before ":"

**Pull request merged**